code/modules/awaymissions/mission_code/beach.dm

```diff
@@ -27,3 +27,154 @@
+	if(!linkedcontroller)
+		return
+	if(ismob(AM))
+		linkedcontroller.mobinpool -= AM
+
+/turf/simulated/floor/beach/away/water/InitializedOn(atom/A)
+	if(!linkedcontroller)
+		return
+	if(istype(A, /obj/effect/decal/cleanable)) // Better a typecheck than looping through thousands of turfs everyday
+		linkedcontroller.decalinpool += A
+
+/turf/simulated/floor/beach/away/water/dense			//for boundary "walls"
+	density = TRUE
+	baseturf = /turf/simulated/floor/beach/away/water/dense
+
+/turf/simulated/floor/beach/away/water/edge_drop
+	name = "Water"
+	icon_state = "seadrop"
+	water_overlay_image = "water_drop"
+	baseturf = /turf/simulated/floor/beach/away/water/edge_drop
+
+/turf/simulated/floor/beach/away/water/drop
+	name = "Water"
+	icon = 'icons/turf/floors/seadrop.dmi'
+	icon_state = "seadrop"
+	water_overlay_image = null
+	smooth = SMOOTH_TRUE
+	canSmoothWith = list(
+	/turf/simulated/floor/beach/away/water/drop, /turf/simulated/floor/beach/away/water/drop/dense,
+		/turf/simulated/floor/beach/away/water, /turf/simulated/floor/beach/away/water/dense,
+		/turf/simulated/floor/beach/away/water/edge_drop)
+	var/obj/effect/beach_drop_overlay/water_overlay
+	baseturf = /turf/simulated/floor/beach/away/water/drop
+
+/turf/simulated/floor/beach/away/water/drop/Initialize(mapload)
+	. = ..()
+	water_overlay = new(src)
+
+/turf/simulated/floor/beach/away/water/drop/Destroy()
+	QDEL_NULL(water_overlay)
+	return ..()
+
+/obj/effect/beach_drop_overlay
+	name = "Water"
+	icon = 'icons/turf/floors/seadrop-o.dmi'
+	layer = MOB_LAYER + 0.1
+	smooth = SMOOTH_TRUE
+	anchored = 1
+	canSmoothWith = list(
+		/turf/simulated/floor/beach/away/water/drop, /turf/simulated/floor/beach/away/water/drop/dense,
+		/turf/simulated/floor/beach/away/water, /turf/simulated/floor/beach/away/water/dense,
+		/turf/simulated/floor/beach/away/water/edge_drop)
+
+/turf/simulated/floor/beach/away/water/drop/dense
+	density = TRUE
+	baseturf = /turf/simulated/floor/beach/away/water/drop/dense
+
+/turf/simulated/floor/beach/away/water/deep
+	name = "Deep Water"
+	icon_state = "seadeep"
+	water_overlay_image = "water_deep"
+	baseturf = /turf/simulated/floor/beach/away/water/deep
+
+/turf/simulated/floor/beach/away/water/deep/dense
+	density = TRUE
+	baseturf = /turf/simulated/floor/beach/away/water/deep/dense
+
+/turf/simulated/floor/beach/away/water/deep/wood_floor
+	name = "Sunken Floor"
+	icon = 'icons/turf/floors.dmi'
+	icon_state = "wood"
+	baseturf = /turf/simulated/floor/beach/away/water/deep/wood_floor
+
+/turf/simulated/floor/beach/away/water/deep/sand_floor
+	name = "Sea Floor"
+	icon_state = "sand"
+	baseturf = /turf/simulated/floor/beach/away/water/deep/sand_floor
+
+/turf/simulated/floor/beach/away/water/deep/rock_wall
+	name = "Reef Stone"
+	icon_state = "desert7"
+	density = TRUE
+	opacity = TRUE
+	explosion_block = 2
+	mouse_opacity = MOUSE_OPACITY_ICON
+	baseturf = /turf/simulated/floor/beach/away/water/deep/rock_wall
+
+/obj/effect/baseturf_helper/beach/away/sand
```

Comment:
away missions and ruins need baseturf helpers because when you play a turf on an existing turf, the new turf inherits the old turf's baseturf---this is what we want in 99% of scenarios to avoid weird oddities, but it does have some unintended consequences for things like ruins and away missions.

In the case of away missions, the z-level is technically `/turf/space`, so any turfs moved to this inherit the space as the baseturf.